x-pack/plugins/alerting/server/rules_client/lib/increment_revision.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObject } from '@kbn/core/server';
+import { get, isEqual } from 'lodash';
+import { RawRule, RuleTypeParams } from '../../types';
+import { fieldsToExcludeFromRevisionUpdates, UpdateOptions } from '..';
+
+export function incrementRevision<Params extends RuleTypeParams>(
+  currentRule: SavedObject<RawRule>,
+  { data }: UpdateOptions<Params>,
+  updatedParams: RuleTypeParams
+): number {
+  // Diff root level fields
+  for (const [field, value] of Object.entries(data).filter(([key]) => key !== 'params')) {
+    if (!fieldsToExcludeFromRevisionUpdates.map(toString).includes(field)) {
+      if (!isEqual(value, get(currentRule.attributes, field))) {
+        return currentRule.attributes.revision + 1;
+      }
+    }
+  }
+
+  // Diff rule params
+  for (const [field, value] of Object.entries(updatedParams)) {
+    // TODO: Should RuleTypes have a way to declare fields that should be revision skipped as well?
```

Comment:
Probably out of scope for this initial PR, but something we haven't really talked about as far as I'm aware. E.g., should O11y or Security Rules be able to specify certain fields that when modified shouldn't result in a revision increment?

Alternatively, should we be inclusive (allow list) instead of exclusive with specifying `fieldsToExcludeFromRevisionUpdates`?

Reply:
++, after working through things today, I think it might actually fit best within the existing `RuleParamsModifier` solutions can provide today. Let's see what the ResponseOps folks have to say and maybe I can add that as an optional param to the existing `RuleParamsModifier` for folks in the future.